src/hooks/core/chat/useMoveChatTypeHook.ts

```diff
@@ -38,42 +39,30 @@ const useMoveChatTypeHook = () => {
       topics: 'TOPIC'
     };
 
-    let signedChannelName;
-    let signedChannelImage;
-
-    if (!isAnonymous) {
-      const myUserData = channel?.members?.find(
-        (member) => member?.user?.id === signedProfileId
-      )?.user;
-      signedChannelName =
-        channel?.channel_type === 4
-          ? `Anonymous ${channel?.anon_user_info_emoji_name}`
-          : getChatName(channel?.name, myUserData?.username ?? myUserData?.name);
-
-      signedChannelImage =
-        channel?.type === 'group' || channel?.type === 'topics'
-          ? channel?.channel_image
-          : channel?.members?.find((member) => member?.user_id !== signedProfileId)?.user?.image ??
-            DEFAULT_PROFILE_PIC_PATH;
-
-      channel.myUserId = signedProfileId;
-    } else {
-      channel.myUserId = undefined;
-    }
+    const channelListInfo = getChannelListInfo(channel, signedProfileId, anonProfileId);
 
-    const chatName = isAnonymous
-      ? await getAnonymousChatName(channel?.members)
-      : {name: signedChannelName, image: signedChannelImage};
+    const anonUserInfo: AnonUserInfo | undefined = {
+      anon_user_info_emoji_name: channelListInfo?.anonUserInfoEmojiName,
+      anon_user_info_emoji_code: channelListInfo?.anonUserInfoEmojiCode,
+      anon_user_info_color_name: channelListInfo?.anonUserInfoColorName,
+      anon_user_info_color_code: channelListInfo?.anonUserInfoColorCode
+    };
+
+    channel.targetName = channelListInfo?.channelName;
+    channel.targetImage = channelListInfo?.channelImage;
 
-    channel.targetName = chatName?.name;
-    channel.targetImage = chatName?.image;
     channel.firstMessage = isAnonymous
       ? channel?.messages?.[0]
       : channel?.messages?.[channel?.messages?.length - 1];
     channel.channel = {...channel};
 
     const channelType = channel?.type;
-    const channelList = ChannelList.fromChannelAPI(channel, type[channelType], channel?.members);
+    const channelList = ChannelList.fromChannelAPI(
+      channel,
+      type[channelType],
+      undefined,
+      anonUserInfo
+    );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The way anonymous user information is handled has been changed significantly. The `getChannelListInfo` function is now used to get the channel list info, which includes anonymous user information. This information is then used to create an `AnonUserInfo` object. Make sure that the `getChannelListInfo` function correctly handles all edge cases and returns valid data. Also, ensure that the `AnonUserInfo` object is correctly used in the rest of the code.



<!-- This is an auto-generated comment by OSS CodeRabbit -->